contracts/Config.sol

```diff
@@ -5,23 +5,25 @@ import "./interfaces/IConfig.sol";
 import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
 
 contract Config is IConfig, OwnableUpgradeable {
-
     event ConfigSetup(address crankAddress, address feeReceiver);
 
     address private crankAddress;
     address private feeReceiver;
+    uint8 private validForDays;
```

Comment:
it feels odd that the state that we hold is the less fine grained time unit. We could still keep the outside API such that when you setup the contract you specify the "validFor" in units of days, but internally we could store as seconds. Be using the more fine grained unit within the contract that prevents the need from having to convert to from days to seconds when performing crank calls--which lowers over all gas usage. There is still the need to convert from days to seconds in the setup, but that would be a one time call.

Reply:
updated